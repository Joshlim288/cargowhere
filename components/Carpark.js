import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Pressable,
  ImageBackground,
  FlatList
} from 'react-native';
import {useSelector} from 'react-redux';

/**
 * @module Carpark
 * */

/**
 * Renders a Carpark Component based on the info available
 * @param {*} props
 * Expected props include, carparkCode, carparkName, availableLots
 */

/**
   *
   * Helper function based on the Haversine Formula to calculate distance between two points using longitude and latitude. 
   * Used to calculate distance between current/specified location and nearby carparks
   * @name getDistanceFromLatLonInM
   * @function
   * @param {Number} lat1 Latitude of the first location
   * @param {Number} lon1 Longitude of the first location
   * @param {Number} lat2 Latitude of the second location
   * @param {Number} lon2 Longitude of the second location
   * @returns {Number} Distance in metres
   */

export function getDistanceFromLatLonInM(lat1, lon1, lat2, lon2) {
  var R = 6371; // Radius of the earth in km
  var dLat = deg2rad(lat2 - lat1);
  var dLon = deg2rad(lon2 - lon1);
  var a =
    Math.sin(dLat / 2) * Math.sin(dLat / 2) +
    Math.cos(deg2rad(lat1)) *
      Math.cos(deg2rad(lat2)) *
      Math.sin(dLon / 2) *
      Math.sin(dLon / 2);
  var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
  var d = R * c * 1000;
  return d;
}

/**
   *
   * Convert values in degree to radian. Helper function to convert values in degree to radian
   * @name deg2rad
   * @function
   * @param {Number} deg Degree Value to be converted
   * @returns {Number} Value in radian
   */

function deg2rad(deg) {
  return deg * (Math.PI / 180);
}

export default function CarparkContainer(props) {
  const availability = useSelector(
    (state) => state.availability.availabilityData,
  );
  const carparks = useSelector(
    (state) => state.carparks.carparksData,
  );
  const maxCarparks = useSelector(
    (state) => state.maxCarparks.limit,
  );
  const specificLocation = useSelector(
    (state) => state.specificLocation,
  );

  /**
   * Calculate distance between the selected carpark and the specified location 
   * @name calculateDistance 
   * @function
   * @param {object} latlng Latitude and longitude of the selected carpark 
   * @returns {Number} Distance in metres
   */

  const calculateDistance = (latlng) => {
    let carparkLatitude = latlng.latitude;
    let carparkLongitude = latlng.longitude;
    return getDistanceFromLatLonInM(
      specificLocation.latlng.latitude,
      specificLocation.latlng.longitude,
      carparkLatitude,
      carparkLongitude
    );
  }

  /**
   * Retrieve the available number of parking lots of the selected carpark
   * @name getAvailableNum 
   * @function
   * @param {String} identifier unique code for the carpark selected 
   * @returns {Number} Number of available lots
   */

  const getAvailableNum = (identifier) => {
    if (availability[identifier]) {
      return availability[identifier].availableLots_car;
    } else {
      return '--';
    }
  };
  
  return (
    <FlatList
      data={carparks.slice(0, maxCarparks)}
      keyExtractor={(item, index) => index.toString()}
      renderItem={({item, index}) => {
        return (
          <Carpark
            carpark={item}
            index={index}
            distanceBetween={calculateDistance(item.latlng)}
            availableNum={getAvailableNum(item.identifier)}
            press={props.press}
          />
        );
      }}
    />
  );
}

function Carpark(props) {
  return (
    <Pressable onPress={props.press(props.carpark)}>
      <View style={styles.container}>
        <View style={styles.markerContainer}>
          <ImageBackground
            style={styles.marker}
            source={require('../images/marker.png')}>
            <Text style={styles.carparkNumber}>{props.index + 1}</Text>
          </ImageBackground>
          <Text style={styles.distance}>
            {Math.round(props.distanceBetween) + 'm'}
          </Text>
        </View>
        <View style={styles.infoContainer}>
          <Text style={styles.name}>{props.carpark.title}</Text>
          <Text style={styles.lots}>Lots available: {props.availableNum}</Text>
        </View>
        <TouchableOpacity style={styles.button}>
        </TouchableOpacity>
      </View>
    </Pressable>
  )
}

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderBottomColor: '#eee',
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
  },
  markerContainer: {
    paddingTop: 5,
    flexDirection: 'column',
    alignSelf: 'flex-start',
    alignItems: 'center',
    marginLeft: 10,
  },
  marker: {
    width: 30,
    height: 30,
    marginTop: 5,
    marginBottom: 3,
    //borderWidth: 1,
  },
  distance: {
    color: '#6a94ff',
    fontSize: 11,
    marginBottom: 3,
    width: 40,
    textAlign: 'center',
  },
  infoContainer: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    marginLeft: 15,
  },
  name: {
    fontSize: 15,
    color: '#494949',
  },
  lots: {
    color: '#777777',
    fontSize: 12,
  },
  carparkNumber: {
    color: 'white',
    textAlign: 'center',
    paddingTop: 3,
    fontSize: 12,
  },
});
