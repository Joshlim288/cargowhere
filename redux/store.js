import {configureStore} from '@reduxjs/toolkit';

import carparksReducer from './carparksSlice';
import regionReducer from './regionSlice';
import specificLocationReducer from './specificLocationSlice';
import sortCriteriaReducer from './sortCriteriaSlice';
import availabilityReducer from './availabilitySlice';
import maxCarparksReducer from './maxCarparksSlice';
import selectedFavouriteReducer from './selectedFavouriteSlice';
import alertReducer from './alertSlice';

// our store will have the states carparks and region, and all updates to these states will be done with their reducers
export default configureStore({
  reducer: {
    carparks: carparksReducer,
    region: regionReducer,
    specificLocation: specificLocationReducer,
    sortCriteria: sortCriteriaReducer,
    availability: availabilityReducer,
    maxCarparks: maxCarparksReducer,
    selectedFavourite: selectedFavouriteReducer,
    alert: alertReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      immutableCheck: false,
      serializableCheck: false,
    }),
});
