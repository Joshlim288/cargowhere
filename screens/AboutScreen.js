import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingVertical: 10,
    paddingHorizontal: 15,
  },
  title: {
    fontSize: 25,
    fontWeight: 'bold',
  },
  body: {
    fontSize: 15,
    color: 'grey',
  },
});

export default function AboutScreen() {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>CarGoWhere</Text>
      <Text />
      <Text style={styles.body}>Need to know where to park your car?</Text>
      <Text />
      <Text style={styles.body}>
        CarGoWhere provides realtime Carpark Availability data for you to plan
        your trips.
      </Text>
    </View>
  );
}
