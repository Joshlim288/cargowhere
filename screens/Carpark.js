import React from 'react';
import {
  View,
  Image,
  Text,
  StyleSheet,
  TouchableOpacity,
  Pressable,
} from 'react-native';
/**
 * Renders a Carpark Component based on the info available
 * @param {*} props
 * Expected props include, carparkCode, carparkName, availableLots
 */

// Haversine formula for calculating distance between two points using longitude and latitude

function getDistanceFromLatLonInM(lat1, lon1, lat2, lon2) {
  var R = 6371; // Radius of the earth in km
  var dLat = deg2rad(lat2 - lat1); // deg2rad below
  var dLon = deg2rad(lon2 - lon1);
  var a =
    Math.sin(dLat / 2) * Math.sin(dLat / 2) +
    Math.cos(deg2rad(lat1)) *
      Math.cos(deg2rad(lat2)) *
      Math.sin(dLon / 2) *
      Math.sin(dLon / 2);
  var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
  var d = R * c * 1000; // Distance in m
  return d;
}

function deg2rad(deg) {
  return deg * (Math.PI / 180);
}

// Still needs images for the button to see more details
export default function Carpark(props) {
  // Get both destination lat/lon and the carpark lat/lon
  const destinationLongitude = props.currentRegion.longitude;
  const destinationLatitude = props.currentRegion.latitude;
  const carparkLongitude = props.carpark.latlng.longitude;
  const carparkLatitude = props.carpark.latlng.latitude;
  const distanceBetween = getDistanceFromLatLonInM(
    destinationLatitude,
    destinationLongitude,
    carparkLatitude,
    carparkLongitude,
  );
  return (
    <Pressable onPress={props.press(props.carpark)}>
      <View style={styles.container}>
        <View style={styles.markerContainer}>
          <Image
            style={styles.marker}
            source={require('../images/marker.png')}
          />
          <Text style={styles.distance}>
            {Math.round(distanceBetween) + 'm'}
          </Text>
        </View>
        <View style={styles.infoContainer}>
          <Text style={styles.name}>{props.carpark.title}</Text>
          {/*should be `${props.carparkName} (${carparkCode})*/}
          <Text style={styles.lots}>Lots available: 20</Text>
          {/*`Lots available ${props.availableLots}`*/}
        </View>
        <TouchableOpacity style={styles.button}>
          {/* <Image source={require('images/chevron-left.png')} /> */}
        </TouchableOpacity>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderBottomColor: '#eee',
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
  },
  markerContainer: {
    paddingTop: 5,
    flexDirection: 'column',
    alignSelf: 'flex-start',
    alignItems: 'center',
    marginLeft: 10,
  },
  marker: {
    width: 20,
    height: 20,
    marginTop: 5,
    marginBottom: 3,
    borderWidth: 1,
  },
  distance: {
    color: '#6a94ff',
    fontSize: 11,
    marginBottom: 3,
    width: 40,
  },
  infoContainer: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    marginLeft: 15,
  },
  name: {
    fontSize: 15,
    color: '#494949',
  },
  lots: {
    color: '#777777',
    fontSize: 12,
  },
});
