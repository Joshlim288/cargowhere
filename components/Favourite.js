import React from 'react';
import {
  View,
  Image,
  Text,
  StyleSheet,
  TouchableOpacity,
  Pressable,
} from 'react-native';

import { useSelector, useDispatch } from 'react-redux';
import { setLatlng } from '../redux/regionSlice';
import { setSpecificLocation } from '../redux/specificLocationSlice';
import { setSelectedFavourite } from '../redux/selectedFavouriteSlice';

export default function Favourite(props) {
  
  const region = useSelector(state => state.region);
  const selectedFavourite = useSelector((state) => state.selectedFavourite);
  const dispatch = useDispatch();

  function onSelectFavourite(){
    dispatch(setLatlng({
      latitude: JSON.parse(props.favourite[1])[0],
      longitude: JSON.parse(props.favourite[1])[1],
    }));
    dispatch(setSpecificLocation({
      latlng: {
        latitude: JSON.parse(props.favourite[1])[0],
        longitude: JSON.parse(props.favourite[1])[1],
      },
      title: props.favourite[0],
    }));
    dispatch(setSelectedFavourite(props.favourite)); 
    props.changeIfNew(false);
}

  return (
    <Pressable> 
      {/* //Might not need this Pressable */}
      <View style={styles.container}>
        <TouchableOpacity style = {styles.touchable}
        onPress = {onSelectFavourite}>
          <View style={styles.markerContainer}>
            <Image
              style={styles.marker}
              source={require('../images/heart.png')}
            />
          </View>
          <View style={styles.infoContainer}>
            <Text style={styles.name}>{props.favourite[0]}</Text>
          </View>
          </TouchableOpacity>
          <TouchableOpacity style={styles.button}
          onPress={() => {
            onSelectFavourite();
            props.changeModalVisible(true);
          }}>
              <Image
                  style={styles.marker}
                  source={require('../images/edit.png')}
              />
        </TouchableOpacity>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderBottomColor: '#eee',
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
    justifyContent: "space-between",
    paddingLeft: 20,
    paddingRight: 35,
  },
  markerContainer: {
    paddingTop: 5,
    flexDirection: 'column',
    alignSelf: 'flex-start',
    alignItems: 'center',
  },
  touchable : {
    flexDirection:"row",
    width: "100%",
  },
  marker: {
    width: 20,
    height: 20,
    marginVertical: 5,
    borderWidth: 1,
  },
  infoContainer: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    marginLeft: 15,
    marginVertical: 8,
  },
  name: {
    fontSize: 15,
    color: '#494949',
  },
});
