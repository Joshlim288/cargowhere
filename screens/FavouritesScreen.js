import React, {useState, useEffect} from 'react';
import {View, Image, Text, Button, FlatList, StyleSheet, Modal, TouchableOpacity} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import Favourite from '../components/Favourite';
import SimpleModal from '../components/SimpleModal';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  header: {
    borderBottomWidth: 1,
    borderBottomColor: '#eee',
    flexDirection: 'row',
    backgroundColor: '#eee',
    paddingVertical: 10,
    justifyContent: "space-between",
  },
  headerText: {
    fontSize: 20,
    paddingLeft: 20,
    textDecorationLine: 'underline',
  },
  favourite: {

  }
});

export default function FavouritesScreen(props) {

  const [isModalVisible, setIsModalVisible] = useState(initialState)
  const [chooseData, setChooseData] = useState();

  const changeModalVisible = (bool) =>{
    setIsModalVisible(bool);
  }

  const setData = (data) => {
    setChooseData(data);
  }

  //Used to load all favourites from local storage, called whenever 
  //favourites are changed or when Favourites tab is clicked
  async function loadAllFavourites() {
    let keys = [];
    let jsonValues = [];
    try {
      keys = await AsyncStorage.getAllKeys();
      jsonValues = await AsyncStorage.multiGet(keys);
      console.log(jsonValues);
      setFavourites(jsonValues);
      //finalAns.push(JSON.parse(value));
    } catch (err) {
      alert(err);
    }
  }

  //Used to add a favourite from local storage
  async function addFavourite(key, value) {
    try {
      const jsonValue = JSON.stringify(value);
      await AsyncStorage.setItem(key, jsonValue);
    } catch (err) {
      alert(err);
    }
    console.log('Executed addFavourite!');
    loadAllFavourites();
  }
  
  //Used to remove a favourite from local storage
  async function removeFavourite(key) {
    try {
      await AsyncStorage.removeItem(key);
      console.log('Removed success');
    } catch (err) {
      alert(err);
    }
    console.log('Executed removeFavourite!');
    loadAllFavourites();
  }

  const [favourites, setFavourites] = useState();

  useEffect(() => {
    //Test adding ======================================================
    addFavourite('Favourites_Key0', 'press above button to remove this!');
    addFavourite("Favourites_Key1", "Tekong");
    addFavourite("Favourites_Key2", "Nanyang Technological University");
    addFavourite("Favourites_Key3", "Plaza Singapura");
    addFavourite("Favourites_Key4", "Sentosa");
    //removeFavourite("Favourites_Key1");
    //==================================================================
    console.log("running useffect");
    loadAllFavourites();
 }, []);

  //TODO
  const manageFavourite = (favourite) => () => {

  };

  return (
    <View style={styles.container}>
      {/* To add CRUD functionality  */}
      <View style={styles.header}>
        <Text style={styles.headerText}>Favourites</Text>
        <TouchableOpacity style = {styles.favourite}
            onPress = {() => changeModalVisible(true)}
            style = {styles.touchableOpacity}
          >
            <Text style = {styles.headerText}>Add To Favourites</Text>
          </TouchableOpacity>
      </View>
      <Modal
      transparent = {true}
      animationType = 'fade'
      visible = {isModalVisible}
      onRequestClose = {() => changeModalVisible(false)}
      >
        <SimpleModal
          changeModalVisible={changeModalVisible}
          setData = {setData}></SimpleModal>
      </Modal>
      {/* Temp button here to test if data can be edited */}
      <Button
        title="Press to remove/add"
        onPress={() => {
          removeFavourite('Favourites_Key0');
        }}
      />
      <FlatList
        style={{width: '100%'}}
        data={favourites}
        keyExtractor={(item, index) => item.key}
        renderItem={({ item }) => (
          <TouchableOpacity
            onPress = {() => changeModalVisible(true)}
            style = {styles.touchableOpacity}
          >
            <Favourite
              favourite={item}
              //currentRegion={region}
              press={manageFavourite}
            />
          </TouchableOpacity>
        )}
      />
    </View>
  );
}
