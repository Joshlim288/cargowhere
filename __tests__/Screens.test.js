/**
 * @format
 */
import React from 'react';
import AboutScreen from '../screens/AboutScreen';
import Carpark from '../screens/Carpark';
import DetailedView from '../screens/DetailedView';
import Favourite from '../screens/Favourite';
import FavouritesScreen from '../screens/FavouritesScreen';
import NearbyScreen from '../screens/NearbyScreen';
import SearchScreen from '../screens/SearchScreen';

import {Provider} from 'react-redux';
import store from '../redux/store';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

/*
let outputData = '';
var storeLog = (inputs) => (outputData += inputs);
console['log'] = jest.fn(storeLog);
console['error'] = jest.fn(storeLog);
These lines of code are so that Jest doesn't throw an error 
when the components attempt to console.log() or console.error(),
especially when there are async functions that will console.log or console.error
after Jest is done testing that it renders correctly
*/

it('AboutScreen renders correctly', () => {
  let outputData = '';
  var storeLog = (inputs) => (outputData += inputs);
  console.log = jest.fn(storeLog);
  console.error = jest.fn(storeLog);
  const tree = renderer.create(<AboutScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Carpark renders correctly', () => {
  let outputData = '';
  var storeLog = (inputs) => (outputData += inputs);
  console.log = jest.fn(storeLog);
  console.error = jest.fn(storeLog);

  var currentRegion = {
    longitude: 0,
    latitude: 0,
  };
  var latlng = {
    longitude: 0,
    latitude: 0,
  };
  var carpark = {latlng};

  const tree = renderer
    .create(
      <Provider store={store}>
        <Carpark
          currentRegion={currentRegion}
          carpark={carpark}
          press={jest.fn()}
        />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('DetailedView renders correctly', () => {
  let outputData = '';
  var storeLog = (inputs) => (outputData += inputs);
  console.log = jest.fn(storeLog);
  console.error = jest.fn(storeLog);
  const tree = renderer
    .create(
      <Provider store={store}>
        <DetailedView />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Favourite renders correctly', () => {
  let outputData = '';
  var storeLog = (inputs) => (outputData += inputs);
  console.log = jest.fn(storeLog);
  console.error = jest.fn(storeLog);

  var favourite = [];
  const tree = renderer
    .create(
      <Provider store={store}>
        <Favourite favourite={favourite} press={jest.fn()} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('FavouritesScreen renders correctly', () => {
  let outputData = '';
  var storeLog = (inputs) => (outputData += inputs);
  console.log = jest.fn(storeLog);
  console.error = jest.fn(storeLog);
  const tree = renderer
    .create(
      <Provider store={store}>
        <FavouritesScreen />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('NearbyScreen renders correctly', () => {
  let outputData = '';
  var storeLog = (inputs) => (outputData += inputs);
  console.log = jest.fn(storeLog);
  console.error = jest.fn(storeLog);
  const tree = renderer
    .create(
      <Provider store={store}>
        <NearbyScreen />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('SearchScreen renders correctly', () => {
  let outputData = '';
  var storeLog = (inputs) => (outputData += inputs);
  console.log = jest.fn(storeLog);
  console.error = jest.fn(storeLog);
  const tree = renderer
    .create(
      <Provider store={store}>
        <SearchScreen />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
