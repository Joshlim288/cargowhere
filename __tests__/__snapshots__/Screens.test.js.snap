// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 25,
        "fontWeight": "bold",
      }
    }
  >
    CarGoWhere
  </Text>
  <Text />
  <Text
    style={
      Object {
        "color": "grey",
        "fontSize": 15,
      }
    }
  >
    Need to know where to park your car?
  </Text>
  <Text />
  <Text
    style={
      Object {
        "color": "grey",
        "fontSize": 15,
      }
    }
  >
    CarGoWhere provides realtime Carpark Availability data for you to plan your trips.
  </Text>
</View>
`;

exports[`Carpark renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#eee",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "column",
          "marginLeft": 10,
          "paddingTop": 5,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "height": 30,
            "marginBottom": 3,
            "marginTop": 5,
            "width": 30,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/marker.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 30,
                "width": 30,
              },
              undefined,
            ]
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 12,
              "paddingTop": 3,
              "textAlign": "center",
            }
          }
        >
          NaN
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#6a94ff",
            "fontSize": 11,
            "marginBottom": 3,
            "textAlign": "center",
            "width": 40,
          }
        }
      >
        0m
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginLeft": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#494949",
            "fontSize": 15,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#777777",
            "fontSize": 12,
          }
        }
      >
        Lots available: 20
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    />
  </View>
</View>
`;

exports[`DetailedView renders correctly 1`] = `null`;

exports[`Favourite renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#eee",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "column",
          "marginLeft": 10,
          "paddingTop": 5,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/heart.png",
          }
        }
        style={
          Object {
            "borderWidth": 1,
            "height": 20,
            "marginBottom": 3,
            "marginTop": 5,
            "width": 20,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginLeft": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#494949",
            "fontSize": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/chevron.png",
          }
        }
        style={
          Object {
            "borderWidth": 1,
            "height": 20,
            "marginBottom": 3,
            "marginTop": 5,
            "width": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`FavouritesScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#eee",
        "borderBottomColor": "#eee",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "paddingLeft": 20,
          "textDecorationLine": "underline",
        }
      }
    >
      Favourites
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press to remove/add
      </Text>
    </View>
  </View>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`NearbyScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": "100%",
        "left": "0%",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#eee",
          "borderBottomColor": "#eee",
          "borderBottomWidth": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "paddingLeft": 20,
            "paddingTop": 10,
            "textDecorationLine": "underline",
          }
        }
      >
        Nearby
      </Text>
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 20,
            "paddingTop": 10,
            "textAlign": "right",
          }
        }
      >
        Sort By:
      </Text>
      <View
        style={
          Object {
            "color": "black",
            "width": "43%",
          }
        }
      >
        <RNCPicker
          items={
            Array [
              Object {
                "label": "Distance",
                "testID": undefined,
                "textColor": undefined,
                "value": "distance",
              },
              Object {
                "label": "Availability",
                "testID": undefined,
                "textColor": undefined,
                "value": "availability",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "height": "100%",
        "left": "100%",
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`SearchScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "lightgrey",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 50,
        "margin": 10,
        "padding": 3,
        "paddingLeft": 20,
        "paddingRight": 20,
        "width": "94%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Search..."
      rejectResponderTermination={true}
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
